README.md
# geany-dark-scheme

![Geany Dark Scheme Icon](./img/geany-dark.png)

This is a dark syntax colouring theme for the [Geany](http://www.geany.org/) text editor.

![Screenshot of Geany Dark Scheme](img/Screenshot-2.png)

Please help us extend the geany-dark theme to support new languages! Please extend the current theme by using the colours from one of the existing syntax files - that way they all play nice together.

## Language Support

The scheme currently supports highlighting for these languages:

- .c - C languages files (contributed by Paulo Cabido)
- .cs - C# language files (contributed by Duncan Lock)
- .cpp - C++ (contributed by gtk.monkey)
- .conf - Apache (and Geany) style .conf files (contributed by Duncan Lock)
- .css - Cascading Stylesheet files (contributed by Roman Snitko)
- .d - d language files (contributed by gtk.monkey)
- .docbook - Docbook mark-up files (contributed by gmunkhbaatarmn)
- .haskel - Haskel code (contributed by Dave.Sarman)
- .html - Hypertext Markup Language file - web pages (contributed by Roman Snitko, Javascript support added by Daniel Mlodecki)
- .java - Java language files (contributed by Paulo Cabido)
- .javascript - Javascript scripts (Duncan Lock, partially based on GeanyDarkTango?)
- .latex - Latex files (contributed by Enrico Tröger)
- .lua - Lua script files (contributed by curtstrangward)
- .markdown - Markdown text files (contributed by pho3nixf1re84)
- .NSIS - Nullsoft Scriptable Installer System scripts (contributed by mr.soup12, modified by Duncan Lock)
- .pascal - Pascal code (contributed by gtk.monkey)
- .perl - Perl language files (contributed by Roman Snitko)
- .php (contributed by Roman Snitko, improved by s5n43K3s)
- .py - Python language files (contributed by Nicolas Hainaux)
- .r - R stats language (contributed by weibullguy)
- .rb - Ruby language files (contributed by Roman Snitko)
- .sh - Bash shell scripts (contributed by Duncan Lock)
- .sql - SQL database query files (contributed by Duncan Lock)
- .yaml - YAML files (contributed by pho3nixf1re84)
- .vala - Vala language files (contributed by boromil)
- .xml - Extensible Markup Language files (contributed by Roman Snitko)

## How to Install the Theme

### Linux: (Ubuntu)

#### Local User (just for you):

The simplest way to do this is to copy the contents of the filedefs folder into the `~/.config/geany/filedefs/` folder. This means that you can only have one theme installed at a time, though.

A better way, is to copy the contents filedefs folder into a `~/.config/geany/geany_dark_filedefs/` folder, creating the folder if it doesn't already exist. Then create a symlink to that folder:

```bash
ln -s ~/.config/geany/geany_dark_filedefs/ ~/.config/geany/filedefs
```

This allows you to have several themes stored in your `~/.config/geany/` folder, and to change which one you're using by changing where the `~/.config/geany/filedefs` symlink points.

#### System Wide:

**BE SURE TO BACK UP THE EXISTING FILEDEFS IF YOU DON'T WANT TO LOSE THEM!**

Copy the filedefs folder into `/usr/share/geany/`

**NOTE**: Geany may be installed in `/usr/local/geany/` or `/opt/geany/` depending on your OS or your installation method. Be sure to choose the correct directory. This method means that your filedef files will get overwritten when you upgrade Geany and also means that you can only have one theme installed at a time.

### Windows:
Copy the contents of the `filedefs` folder to...

**Current User only:**

`%APPDATA%\geany\filedefs`

**System Wide:**

**BE SURE TO BACK UP THE EXISTING FILEDEFS IF YOU DON'T WANT TO LOSE THEM!**

`C:\Program Files\Geany\data`

NOTE: This method means that your filedef files will get overwritten when you upgrade Geany and also means that you can only have one theme installed at a time.


CHANGELOG.md
# geany-dark-scheme Changelog

### 0.1 - inital release after migration to github

- Ran the google code exported tool, which migrated over the wiki articles and the issues.
- Manually moved all assets over to github, converted wiki pages to readme/changelog markdown files and checked everything into git.

## Google Code hosted Project

### geany_dark_filedefs_20100304_190847.tar.bz2
+ Added support for Markdown - .markdown (contributed by pho3nixf1re84, modified by Duncan Lock). I added the compiler option to this 'compiler=markdown "%f" > "%e".html' which will attempt compile your markdown file to filename.html using the default perl markdown script from John Gruber. I also set the default extension to .markdown
+ Added support for Nullsoft Scriptable Installer System scripts - .nsis (contributed by mr.soup12, modified by Duncan Lock). I adjusted the colours to fit the theme colours better.

### geany_dark_filedefs_20091119_234556.tar.bz2
+ Added support for Lua - .lua (contributed by curtstrangward)

### geany_dark_filedefs_20090927_201349.tar.bz2
+ Added support for Vala - .vala (contributed by boromil)
+ Added support for Docbook - .docbook (contributed by gmunkhbaatarmn)
+ Added support for yaml - .yaml (contributed by pho3nixf1re84)

### geany_dark_filedefs_20090516_193911.tar.bz2
+ Improved javascript support from etienne.deparis & ludwig.arne
+ Improved javascript in HTML support from T.Karbownicki

### geany_dark_filedefs_20090516_190439.tar.bz2
+ Added quick-and-dirty javascript support by Duncan Lock, partially based on (http://www.barryvan.com.au/2009/01/geany-ide-tango-dark-colour-scheme/)

### geany_dark_filedefs_20090516_182543.tar.bz2
+ Improved .php support - .php (improved by s5n43K3s)

### geany_dark_filedefs_20090516_181549.tar.bz2
+ Added support for C++ - .cpp - C++ (contributed by gtk.monkey) 
+ Added support for D - .d language files (contributed by gtk.monkey)
+ Added support for Haskel - .haskel (contributed by Dave.Sarman)
+ Added support for R stats language - .r (contributed by weibullguy) 

### geany_dark_filedefs_20081209_215628.tar.bz2
+ Added support for Latex - .latex page layout language (added by Enrico Tröger)

### geany_dark_filedefs_20081111_185356.tar.bz2
+ Added support for C# - .cs (can't remember where this came from)
+ Added support for c - .c (can't remember where this came from)

### geany_dark_filedefs_20080906_175251.tar.bz2
+ Initial version, including support for sql, sh, perl, conf, python, java, css, ruby & html - Most of these were originally developed by Roman Snitko, with contributions from Daniel Mlodecki (Javascript support in HTML), Paulo Cabido (C & Java), Nicolas Hainaux (Python) and me (C#, conf, sh & sql)


filedefs/filetypes.c
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
commentline=0x747474;0x1E1E1E;false;false
comment=0x747474;0x1E1E1E;false;false
commentdoc=0x747474;0x1E1E1E;false;false

number=0x7ECDFF;0x1E1E1E;false;false

word=0xFFCB4F;0x1E1E1E;true;false

#structs type
#word2=0xA843E6;0x1E1E1E;true;false
#word2=0x7EB35B;0x1E1E1E;true;false
word2=0xAAFF57;0x1E1E1E;true;false

string=0xA18651;0x1E1E1E;true;false
character=0xA18651;0x1E1E1E;true;false

#guids
uuid=0xffffff;0x1E1E1E;false;false

#includes
preprocessor=0xB2A886;0x1E1E1E;true;false

operator=0x98BAC5;0x1E1E1E;true;false
#operator=0x7EB35B;0x1E1E1E;true;false
identifier=0xffffff;0x1E1E1E;false;false

#end of line where string is not closed
stringeol=0xffffff;0xffffff;true;false

#string @"str"
verbatim=0xDBB260;0x1E1E1E;false;false

regex=0xAAFF57;0x1E1E1E;false;false

commentlinedoc=0x747474;0x1E1E1E;false;false
commentdockeyword=0xffffff;0x1E1E1E;false;false
commentdockeyworderror=0x747474;0x1E1E1E;false;false

#array index
globalclass=0x7EB35B;0x1E1E1E;false;false

styling_within_preprocessor=1;0;false;false

[keywords]
# all items must be in one line
primary=volatile default export goto sizeof typename asm enum bool union NULL FALSE TRUE typedef struct char int float double void unsigned signed long short extern static register auto const if else switch for while do break continue return class new delete inline case

docComment=attention author brief bug class code date def enum example exception file fn namespace note param remarks return see since struct throw todo typedef var version warning union

[settings]
# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=/*
comment_close=*/

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
	#command_example();
# setting to false would generate this
#	command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
compiler=gcc -Wall -c "%f"
# the -o option is automatically added by Geany
linker=gcc -Wall "%f"
run_cmd="./%e"


filedefs/filetypes.d
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
comment=0xA0A0A0;0x1E1E1E;false;false
commentline=0xA0A0A0;0x1E1E1E;false;false
commentdoc=0xA0A0A0;0x1E1E1E;false;false
commentdocnested=0xA0A0A0;0x1E1E1E;false;false
number=0x7ECDFF;0x1E1E1E;false;false
word=0xFFCB4F;0x1E1E1E;true;false
word2=0xAAFF57;0x1E1E1E;true;false
word3=0xAAFF57;0x1E1E1E;true;false
typedef=0xFFCB4F;0x1E1E1E;true;true
string=0xA18651;0x1E1E1E;true;false
stringeol=0xA18651;0x1E1E1E;true;true
character=0xA18651;0x1E1E1E;true;false
operator=0x98BAC5;0x1E1E1E;true;false
identifier=0xffffff;0x1E1E1E;false;false
commentlinedoc=0xA0A0A0;0x1E1E1E;false;false
commentdockeyword=0xffffff;0x1E1E1E;false;false
commentdockeyworderror=0xA0A0A0;0x1E1E1E;false;false

[keywords]
# all items must be in one line
primary=__FILE__ __LINE__ __DATA__ __TIME__ __TIMESTAMP__ abstract alias align asm assert auto body bool break byte case cast catch cdouble cent cfloat char class const continue creal dchar debug default delegate delete deprecated do double else enum export extern false final finally float for foreach function goto idouble if ifloat import in inout int interface invariant ireal is long mixin module new null out override package pragma private protected public real return scope short static struct super switch synchronized template this throw true try typedef typeof ubyte ucent uint ulong union unittest ushort version void volatile wchar while with
secondary=
# documentation keywords for D, currently not working
docComment=Authors Bugs Copyright Date Deprecated Examples History License Macros Params Returns See_Also Standards Throws Version
types=

[settings]
# default extension used when saving files
#extension=d

# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=//
comment_close=
# this is an alternative way, so multiline comments are used
#comment_open=/*
#comment_close=*/
# this is alternative way, so multiline comments are used
#comment_open=/+
#comment_close=+/

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
        #command_example();
# setting to false would generate this
#       command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
compiler=dmd -w -c "%f"
linker=dmd -w -of"%e" "%f"
# you can also use the gdc compiler, please use the "gdmd" wrapper script(included with gdc)
#compiler=gdmd -w -c "%f"
#linker=gdmd -w -of"%e" "%f"

run_cmd="./%e"

filedefs/filetypes.cs
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false

commentline=0x747474;0x1E1E1E;false;false
comment=0x747474;0x1E1E1E;false;false
commentdoc=0x747474;0x1E1E1E;false;false
number=0x7ECDFF;0x1E1E1E;false;false
word=0xFFCB4F;0x1E1E1E;true;false
word2=0xAAFF57;0x1E1E1E;true;false
string=0xA18651;0x1E1E1E;true;false
character=0xA18651;0x1E1E1E;true;false

#GUID's
uuid=0xffffff;0x1E1E1E;false;false

#includes
preprocessor=0xB2A886;0x1E1E1E;true;false

operator=0x98BAC5;0x1E1E1E;true;false
#operator=0x7EB35B;0x1E1E1E;true;false
identifier=0xffffff;0x1E1E1E;false;false

#end of line where string is not closed
stringeol=0xffffff;0xffffff;true;false

#string @"str"
verbatim=0xDBB260;0x1E1E1E;false;false

regex=0xAAFF57;0x1E1E1E;false;false

commentlinedoc=0x747474;0x1E1E1E;false;false
commentdockeyword=0xffffff;0x1E1E1E;false;false
commentdockeyworderror=0x747474;0x1E1E1E;false;false

#array index
globalclass=0x7EB35B;0x1E1E1E;false;false

# whether arguments of preprocessor commands should be styled (only first argument is used)
# 1 to enable, 0 to disable
styling_within_preprocessor=1;0;false;false

[keywords]
# all items must be in one line
primary=abstract as base bool break byte case catch char checked class const continue decimal default delegate do double else enum event explicit extern false finally fixed float for foreach goto if implicit in int interface internal is lock long namespace new null object operator out override params private protected public readonly ref return sbyte sealed short sizeof stackalloc static string struct switch this throw true try typeof uint ulong unchecked unsafe ushort using virtual void volatile while
secondary=
# these are some doxygen keywords (incomplete)
docComment=attention author brief bug class code date def enum example exception file fn namespace note param remarks return see since struct throw todo typedef var version warning union

[settings]
# default extension used when saving files
#extension=cs

# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=//
comment_close=
# this is an alternative way, so multiline comments are used
#comment_open=/*
#comment_close=*/

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
	#command_example();
# setting to false would generate this
#	command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
# be careful following settings are untested
compiler=mcs /t:winexe "%f" /r:System,System.Drawing
run_cmd=mono "%e.exe"


filedefs/filetypes.sh
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
commentline=0x747474;0x1E1E1E;false;false
number=0x7ECDFF;0x1E1E1E;false;false
string=0xA18651;0x1E1E1E;false;false
character=0xA18651;0x1E1E1E;false;false
word=0xFFCB4F;0x1E1E1E;true;false
operator=0x98BAC5;0x1E1E1E;false;false
identifier=0xffffff;0x1E1E1E;false;false
backticks=0x30FF00;0x504250;false;false
param=0x009f00;0x1E1E1E;false;false
scalar==0x7ECDFF;0x1E1E1E;false;false




[keywords]
primary=break case continue do done elif else esac eval exit export fi for function goto if in integer return set shift then until while


[settings]
# default extension used when saving files
#extension=sh

# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=#
comment_close=

# set to false if a comment character/string should start a column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
	#command_example();
# setting to false would generate this
#	command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
run_cmd="./%f"


filedefs/filetypes.cpp
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
commentline=0xA0A0A0;0x1E1E1E;false;false
comment=0xA0A0A0;0x1E1E1E;false;false
commentdoc=0xA0A0A0;0x1E1E1E;false;false
number=0x7ECDFF;0x1E1E1E;false;false
word=0xFFCB4F;0x1E1E1E;true;false
word2=0xAAFF57;0x1E1E1E;true;false
string=0xA18651;0x1E1E1E;true;false
character=0xA18651;0x1E1E1E;true;false
uuid=0xffffff;0x1E1E1E;false;false
preprocessor=0xB2A886;0x1E1E1E;true;false
operator=0x98BAC5;0x1E1E1E;true;false
identifier=0xffffff;0x1E1E1E;false;false
stringeol=0xffffff;0xffffff;true;false
verbatim=0xDBB260;0x1E1E1E;false;false
regex=0xAAFF57;0x1E1E1E;false;false
commentlinedoc=0xA0A0A0;0x1E1E1E;false;false
commentdockeyword=0xffffff;0x1E1E1E;false;false
commentdockeyworderror=0xA0A0A0;0x1E1E1E;false;false
globalclass=0x7EB35B;0x1E1E1E;false;false
# whether arguments of preprocessor commands should be styled (only first argument is used)
# 1 to enable, 0 to disable
styling_within_preprocessor=1;0;false;false

[keywords]
# all items must be in one line
primary=and and_eq asm auto bitand bitor bool break case catch char class compl const const_cast continue default delete do double dynamic_cast else enum explicit export extern false float for friend goto if inline int long mutable namespace new not not_eq operator or or_eq private protected public register reinterpret_cast return short signed sizeof static static_cast struct switch template this throw true try typedef typeid typename union unsigned using virtual void volatile wchar_t while xor xor_eq
secondary=
# these are some doxygen keywords (incomplete)
docComment=attention author brief bug class code date def enum example exception file fn namespace note param remarks return see since struct throw todo typedef var version warning union

[settings]
# default extension used when saving files
#extension=cpp

# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=//
comment_close=
# this is an alternative way, so multiline comments are used
#comment_open=/*
#comment_close=*/

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
        #command_example();
# setting to false would generate this
#       command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
compiler=g++ -Wall -c "%f"
linker=g++ -Wall -o "%e" "%f"
run_cmd="./%e"

filedefs/filetypes.lua
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
comment=0x747474;0x1E1E1E;false;false
commentline=0x747474;0x1E1E1E;false;false
commentdoc=0x747474;0x1E1E1E;false;false
number=0x7ECDFF;0x1E1E1E;false;false
word=0xFFCB4F;0x1E1E1E;true;false
string=0xA18651;0x1E1E1E;false;false
character=0xA18651;0x1E1E1E;false;false
preprocessor=0xB2A886;0x1E1E1E;true;false
operator=0x98BAC5;0x1E1E1E;false;false
identifier=0xffffff;0x1E1E1E;false;false
stringeol=0xA18651;0x1E1E1E;false;false

literalstring=0xA18651;0x1E1E1E;false;false

function_basic=0x729fcf;0x1E1E1E;true;false
function_other=0x729fcf;0x1E1E1E;true;false
coroutines=0x729fcf;0x1E1E1E;true;false
word5=0xFFCB4F;0x1E1E1E;true;false
word6=0xFFCB4F;0x1E1E1E;true;false
word7=0xFFCB4F;0x1E1E1E;true;false
word8=0xFFCB4F;0x1E1E1E;true;false

[keywords]
# all items must be in one line
keywords=and break do else elseif end false for function if in local nil not or repeat return then true until while
# Basic functions
function_basic=_ALERT assert call collectgarbage coroutine debug dofile dostring error _ERRORMESSAGE foreach foreachi _G gcinfo getfenv getmetatable getn globals _INPUT io ipairs load loadfile loadlib loadstring math module newtype next os _OUTPUT pairs pcall print _PROMPT rawequal rawget rawset require select setfenv setmetatable sort _STDERR _STDIN _STDOUT string table tinsert tonumber tostring tremove type unpack _VERSION xpcall
# String, (table) & math functions
function_other=abs acos asin atan atan2 ceil cos deg exp floor format frexp gsub ldexp log log10 math.abs math.acos math.asin math.atan math.atan2 math.ceil math.cos math.cosh math.deg math.exp math.floor math.fmod math.frexp math.huge math.ldexp math.log math.log10 math.max math.min math.mod math.modf math.pi math.pow math.rad math.random math.randomseed math.sin math.sinh math.sqrt math.tan math.tanh max min mod rad random randomseed sin sqrt strbyte strchar strfind string.byte string.char string.dump string.find string.format string.gfind string.gmatch string.gsub string.len string.lower string.match string.rep string.reverse string.sub string.upper strlen strlower strrep strsub strupper table.concat table.foreach table.foreachi table.getn table.insert table.maxn table.remove table.setn table.sort tan
# (coroutines), I/O & system facilities
coroutines=appendto clock closefile coroutine.create coroutine.resume coroutine.running coroutine.status coroutine.wrap coroutine.yield date difftime execute exit flush getenv io.close io.flush io.input io.lines io.open io.output io.popen io.read io.stderr io.stdin io.stdout io.tmpfile io.type io.write openfile os.clock os.date os.difftime os.execute os.exit os.getenv os.remove os.rename os.setlocale os.time os.tmpname package.cpath package.loaded package.loadlib package.path package.preload package.seeall read readfrom remove rename seek setlocale time tmpfile tmpname write writeto
# user definable keywords
user1=
user2=
user3=
user4=

[settings]
# default extension used when saving files
extension=lua

# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=--
comment_close=
# this is an alternative way, so multiline comments are used
#comment_open=--[[
#comment_close=]]--

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
        #command_example();
# setting to false would generate this
#       command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
compiler=
run_cmd=lua "%f"

filedefs/filetypes.java
# For complete documentation of this file, please see Geany's main documentation
[styling]
# foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
comment=0x747474;0x1E1E1E;false;false
commentline=0x747474;0x1E1E1E;false;false
commentdoc=0x747474;0x1E1E1E;false;false

number=0x7ECDFF;0x1E1E1E;false;false

word=0xFFCB4F;0x1E1E1E;true;false
#
word2=0xAAFF57;0x1E1E1E;true;false

string=0xA18651;0x1E1E1E;true;false
character=0xA18651;0x1E1E1E;true;false

#?
uuid=0xffffff;0xffffff;false;false

#includes, defines, etc
preprocessor=0xB2A886;0x1E1E1E;true;false

operator=0x98BAC5;0x1E1E1E;true;false
#operator=0x7EB35B;0x1E1E1E;true;false
identifier=0xffffff;0x1E1E1E;false;false

#end of line where string is not closed
stringeol=0xffffff;0xffffff;true;false

#string @"str"	
verbatim=0xDBB260;0x1E1E1E;false;false

regex=0xAAFF57;0x1E1E1E;false;false

commentlinedoc=0x747474;0x1E1E1E;false;false
commentdockeyword=0xffffff;0x1E1E1E;false;false
commentdockeyworderror=0x747474;0x1E1E1E;false;false
	
#array index
globalclass=0x7EB35B;0x1E1E1E;false;false

styling_within_preprocessor=1;0;false;false

[keywords]
# all items must be in one line
primary=abstract assert break case catch class const continue default do else extends final finally for future generic goto if implements import inner instanceof interface native new outer package private protected public rest return static super switch synchronized this throw throws transient try var volatile while true false null
#added String and Vector to the original
secondary=boolean byte char double float int long null short void String Vector

docComment=attention author brief bug class code date def enum example exception file fn namespace note param remarks return see since struct throw todo typedef var version warning union

[settings]
# the following characters are these which a "word" can contains, see documentation
#wordchars=_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
comment_open=/*
comment_close=*/

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
	#command_example();
# setting to false would generate this
#	command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=

[build_settings]
# %f will be replaced by the complete filename
# %e will be replaced by the filename without extension
# (use only one of it at one time)
compiler=javac "%f"
run_cmd=java "%e"


filedefs/filetypes.r
# For complete documentation of this file, please see Geany's main documentation
[styling]
#foreground;background;bold;italic
default=0xffffff;0x1E1E1E;false;false
comment=0x747474;0x1E1E1E;false;false
kword=0xFFCB4F;0x1E1E1E;true;false
operator=0x98BAC5;0x1E1E1E;true;false
basekword=0xFFCB4F;0x1E1E1E;true;false
otherkword=0xFFCB4F;0x1E1E1E;true;false
number=0x7ECDFF;0x1E1E1E;false;false
string=0xA18651;0x1E1E1E;false;false
string2=0xA18651;0x1E1E1E;false;false
identifier=0xffffff;0x1E1E1E;false;false
infix=0x808000;0x1E1E1E;false;false
infixeol=0x000000;0xe0c0e0;false;false

[keywords] # all items must be in one line
primary=abs array break data.frame diag else FALSE for function if in Inf matrix NA NaN NCOL next NROW NULL print read.table repeat require return solve source sqrt sum TRUE while package=abbreviate abline abs acf acos acosh addmargins aggregate agrep alarm alias alist all anova any aov aperm append apply approx approxfun apropos ar args arima array arrows asin asinh assign assocplot atan atanh attach attr attributes autoload autoloader ave axis backsolve barplot basename beta bindtextdomain binomial biplot bitmap bmp body box boxplot bquote break browser builtins bxp by bzfile c call cancor capabilities casefold cat category cbind ccf ceiling character charmatch chartr chol choose chull citation class close cm cmdscale codes coef coefficients col colnames colors colorspaces colours comment complex confint conflicts contour contrasts contributors convolve cophenetic coplot cor cos cosh cov covratio cpgram crossprod cummax cummin cumprod cumsum curve cut cutree cycle data dataentry date dbeta dbinom dcauchy dchisq de debug debugger decompose delay deltat demo dendrapply density deparse deriv det detach determinant deviance dexp df dfbeta dfbetas dffits dgamma dgeom dget dhyper diag diff diffinv difftime digamma dim dimnames dir dirname dist dlnorm dlogis dmultinom dnbinom dnorm dotchart double dpois dput drop dsignrank dt dump dunif duplicated dweibull dwilcox eapply ecdf edit effects eigen emacs embed end environment eval evalq example exists exp expression factanal factor factorial family fft fifo file filter find fitted fivenum fix floor flush for force formals format formula forwardsolve fourfoldplot frame frequency ftable function gamma gaussian gc gcinfo gctorture get getenv geterrmessage gettext gettextf getwd gl glm globalenv gray grep grey grid gsub gzcon gzfile hat hatvalues hcl hclust head heatmap help hist history hsv httpclient iconv iconvlist identical identify if ifelse image influence inherits integer integrate interaction interactive intersect invisible isoreg jitter jpeg julian kappa kernapply kernel kmeans knots kronecker ksmooth labels lag lapply layout lbeta lchoose lcm legend length letters levels lfactorial lgamma library licence license line lines list lm load loadhistory loadings local locator loess log logb logical loglin lowess ls lsfit machine mad mahalanobis makepredictcall manova mapply match matlines matplot matpoints matrix max mean median medpolish menu merge message methods mget min missing mode monthplot months mosaicplot mtext mvfft names napredict naprint naresid nargs nchar ncol next nextn ngettext nlevels nlm nls noquote nrow numeric objects offset open optim optimise optimize options order ordered outer pacf page pairlist pairs palette par parse paste pbeta pbinom pbirthday pcauchy pchisq pdf pentagamma person persp pexp pf pgamma pgeom phyper pi pico pictex pie piechart pipe plclust plnorm plogis plot pmatch pmax pmin pnbinom png pnorm points poisson poly polygon polym polyroot postscript power ppoints ppois ppr prcomp predict preplot pretty princomp print prmatrix prod profile profiler proj promax prompt provide psigamma psignrank pt ptukey punif pweibull pwilcox q qbeta qbinom qbirthday qcauchy qchisq qexp qf qgamma qgeom qhyper qlnorm qlogis qnbinom qnorm qpois qqline qqnorm qqplot qr qsignrank qt qtukey quantile quarters quasi quasibinomial quasipoisson quit qunif quote qweibull qwilcox rainbow range rank raw rbeta rbind rbinom rcauchy rchisq readline real recover rect reformulate regexpr relevel remove reorder rep repeat replace replicate replications require reshape resid residuals restart return rev rexp rf rgamma rgb rgeom rhyper rle rlnorm rlogis rm rmultinom rnbinom rnorm round row rownames rowsum rpois rsignrank rstandard rstudent rt rug runif runmed rweibull rwilcox sample sapply save savehistory scale scan screen screeplot sd search searchpaths seek segments seq sequence serialize setdiff setequal setwd shell sign signif sin single sinh sink smooth solve sort source spectrum spline splinefun split sprintf sqrt stack stars start stderr stdin stdout stem step stepfun stl stop stopifnot str strftime strheight stripchart strptime strsplit strtrim structure strwidth strwrap sub subset substitute substr substring sum summary sunflowerplot supsmu svd sweep switch symbols symnum system t table tabulate tail tan tanh tapply tempdir tempfile termplot terms tetragamma text time title toeplitz tolower topenv toupper trace traceback transform trigamma trunc truncate try ts tsdiag tsp typeof unclass undebug union unique uniroot unix unlink unlist unname unserialize unsplit unstack untrace unz update upgrade url var varimax vcov vector version vi vignette warning warnings weekdays weights which while window windows with write wsbrowser xedit xemacs xfig xinch xor xtabs xyinch yinch zapsmall package_other=acme aids aircondit amis aml banking barchart barley beaver bigcity boot brambles breslow bs bwplot calcium cane capability cav censboot channing city claridge cloth cloud coal condense contourplot control corr darwin densityplot dogs dotplot ducks empinf envelope environmental ethanol fir frets gpar grav gravity grob hirose histogram islay knn larrows levelplot llines logit lpoints lsegments lset ltext lvqinit lvqtest manaus melanoma melanoma motor multiedit neuro nitrofen nodal ns nuclear oneway parallel paulsen poisons polar qq qqmath remission rfs saddle salinity shingle simplex singer somgrid splom stripplot survival tau tmd tsboot tuna unit urine viewport wireframe wool xyplot

[settings]
# default extension used when saving files
extension=R

# the following characters are these which a "word" can contains, see documentation #
#wordchars=abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789

# if only single comment char is supported like # in this file, leave comment_close blank
#comment_open=
#comment_close=

# set to false if a comment character/string should start at column 0 of a line, true uses any
# indentation of the line, e.g. setting to true causes the following on pressing CTRL+d
        #command_example();
# setting to false would generate this
#       command_example();
# This setting works only for single line comments
comment_use_indent=true

# context action command (please see Geany's main documentation for details)
context_action_cmd=